lib/instance.js

```diff
@@ -642,14 +643,14 @@ Instance.prototype.save = function(options) {
 
       if (self.isNewRecord) {
         query = 'insert';
-        args = [self, self.Model.getTableName(options), values, options];
+        args = [self, self.Model.getTableName(this.$options), values, options];
       } else {
         var where = this.where();
 
         where = Utils.mapValueFieldNames(where, Object.keys(where), this.Model);
 
         query = 'update';
-        args = [self, self.Model.getTableName(options), values, where, options];
+        args = [self, self.Model.getTableName(this.$options), values, where, options];
```

Comment:
`options` is used by end-users, as i wrote in my previous comment, changing it would mean a major bump and we're not looking to do that quite yet :)

`> The issue here is that this.Model is attached to the prototype of the model and not the model instance itself

Not sure what you mean? It's attached to the instance prototype which means its attached to ALL instances. (Not sure how it's relevant either).